presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentControlUtils.java

```diff
@@ -0,0 +1,220 @@
+                    }
+                    String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(entity((new Gson().toJson("SHUTTING_DOWN")), APPLICATION_JSON_TYPE)).buildPut().invoke();
+                    if (response.getStatus() != 200) {
+                        throw new PrestoManagerException("Failed to stop presto gracefully");
+                    }
+                }
+                catch (IOException | URISyntaxException e) {
+                    throw new PrestoManagerException("Error in stopping presto", e);
+                }
+        }
+    }
+
+    public static Response restartPresto()
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).build();
+        }
+        try {
+            int prestoRestart = executeCommand("service", "presto", "restart");
+            if (prestoRestart != 0) {
+                LOGGER.error("Failed to restart presto; Process exited with return value: {}", prestoRestart);
+            }
+        }
+        catch (PrestoManagerException  e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    public static Response prestoStatus()
+    {
+        if (!isInstalled()) {
+            return Response.status(OK).entity(new Gson().toJson("Presto is not installed"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+        if (!isRunning()) {
+            return Response.status(OK).entity(new Gson().toJson("Presto is installed but not running"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+        try {
+            String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+            return newRequestBuilder(new URI(format("localhost:%s/v1/info", prestoPort))).buildGet().invoke();
+        }
+        catch (IOException | URISyntaxException e) {
+            LOGGER.error("Failed to get status", e);
+            return Response.status(INTERNAL_SERVER_ERROR).entity(new Gson().toJson("Failed to get status"))
+                    .type(APPLICATION_JSON_TYPE).build();
+        }
+    }
+
+    private static boolean isInstalled()
+    {
+        try {
+            if (executeCommand("rpm", "-q", "presto-server-rpm") != 0) {
+                LOGGER.error("Presto is not installed");
+                return false;
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+        }
+        catch (Exception e) {
```

Comment:
If there are certain exceptions that can be thrown here that you don't handle, why not let them go?

Most library functions document exactly which exceptions they can throw, so there's little risk of unexpected exceptions.